lib/pure/parseutils.nim

```diff
@@ -87,14 +91,18 @@ proc parseOct*(s: string, number: var int, start = 0): int  {.
     inc(i)
   if foundDigit: result = i-start
 
-proc parseBin*(s: string, number: var int, start = 0): int  {.
+proc parseBin*(s: string, number: var int, start = 0, maxLen = 0): int  {.
   rtl, extern: "npuParseBin", noSideEffect.} =
-  ## parses an binary number and stores its value in ``number``. Returns
+  ## Parses an binary number and stores its value in ``number``. Returns
   ## the number of the parsed characters or 0 in case of an error.
+  ##
+  ## If 'maxLen==0' the length of the binary number has no
+  ## upper bound. Not more than ```maxLen`` characters are parsed.
   var i = start
   var foundDigit = false
-  if i+1 < s.len and s[i] == '0' and (s[i+1] == 'b' or s[i+1] == 'B'): inc(i, 2)
-  while i < s.len:
+  let last = if maxLen == 0: s.len else: i+maxLen  
```

Comment:
What if the passed in maximum length is greater than the string length?

Reply:
That's a very good point. I used the same code as for `parseHex`, without checking it for any bugs (there are no test cases for parseutils, as far as I can tel...). I'll go ahead and fix that for all three.